### ISSUE TITLE
Module defaults groups

### ISSUE THREAD

**Issue by markgraf** [theforeman/foreman-ansible-modules#1015]
##### SUMMARY
As it is now, there are 54 modules that all want
- user
- password
- server-url

Repeating those same informations over and over again with all the modules you use is lots of redundant work.
Creating "module defaults" for each of them is equally redundant.

##### FEATURE REQUEST
 It would be nice to have [Module defaults groups](https://docs.ansible.com/ansible/latest/user_guide/playbooks_module_defaults.html#module-defaults-groups) in order to define that block ONCE for an entire playbook like this:

```
# example_play.yml
- hosts: localhost
  module_defaults:
    group/foreman:
      name: myNameIs
      password: TopS3cr3t
      server_url: foreman.at.home
```
These already exist for aws, azure, gcp, etc..., so I guess most of the needed code should already exist?

**Comment by evgeni** [comment]
This is currently not possible for collections: https://github.com/ansible/ansible/issues/72006
But yes, we also really want to have that feature :)

**Comment by markgraf** [comment]
Good to know.  Thanks for the update!  Waiting for https://github.com/ansible/ansible/issues/72006 then...

**Comment by evgeni** [comment]
Seems this is finally coming in core 2.12: https://github.com/ansible/community/issues/546#issuecomment-832773914

**Comment by markgraf** [comment]
Hey, that's good news! Thanks!

Am April 13, 2022 6:52:04 PM UTC schrieb Evgeni Golov ***@***.***>:
>Closed #1015 via #1344.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/theforeman/foreman-ansible-modules/issues/1015#event-6429638147
>You are receiving this because you authored the thread.
>
>Message ID:
***@***.***>

-- Gesendet von /e/ Mail.
